### ISSUE TITLE
Interacting with the Sphinx Generated Docs from the Landing Page

### ISSUE THREAD

**Issue by littlewonder** [astropy/astropy-tutorials#216]
I was dabbling with the dynamic search integration of the Sphinx Generated pages with the landing page for the learn astropy website. [Issue #203]

I figured out that we can directly make request to [Read the Docs](https://readthedocs.org/api/v2/docsearch/?q=plot&project=astropy-tutorials&version=stable&language=en) to get search results, corresponding to our query (which is plot, in the example URL stated above). But Read the Docs has disabled CORS. So I can’t make a request directly from the frontend.

There’re 2 possible solutions, I am looking at:
- Host a Flask Server to make the HTTP Request: Solves the Issue of CORS but since it’s mentioned in org chart of learn astropy that it needs to be hosted on gh-pages, I don’t know how feasible will be the integration of a backend.
- An uglier solution would be to generate a index (JSON) every time the docs are generated and request that index file to return the query results.

I’d like to have suggestions on how shall I proceed.

**Comment by kelle** [comment]
I don't know what CORS is but is it the kind of thing that we could ask RTD to implement for us? 

@adrn , thoughts?

**Comment by littlewonder** [comment]
As far as I know, enabling CORS(Cross-Origin Resource Sharing) is not a good practice since allowing other websites to do HTTP calls from JavaScript (what we call) can be a security issue; so it's possible that RTD may not agree to our request. Maybe we can ask them to whitelist our particular domain, but still, that depends on them. 

Let's see what @adrn has to say. Meanwhile I'll brainstorm for better alternatives. Using a simple flask based server can be an option.

**Comment by pllim** [comment]
Yeah, let's not get too fancy there with JS lest browser start blocking it due to suspected security risk...

**Comment by littlewonder** [comment]
Problem is If we host a backend(flask or something, to make the HTTP Request), then we can't use Github Pages to host the website. We will need to deploy it somewhere(Heroku or Digital Ocean). Is that possible?
